File: MemfaultPackages/reporting-libs/clients/maven-local-java/src/main/java/com/memfault/bort/reporting/TestReceiver.java
package com.memfault.bort.reporting;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import com.memfault.bort.java.reporting.Distribution;
import com.memfault.bort.java.reporting.Reporting;
import java.util.List;

import static com.memfault.bort.reporting.NumericAgg.COUNT;
import static com.memfault.bort.reporting.NumericAgg.MAX;
import static com.memfault.bort.reporting.NumericAgg.MEAN;
import static com.memfault.bort.reporting.NumericAgg.MIN;
import static com.memfault.bort.reporting.NumericAgg.SUM;
import static com.memfault.bort.reporting.StateAgg.TIME_PER_HOUR;
import static com.memfault.bort.reporting.StateAgg.TIME_TOTALS;
import static java.util.Arrays.asList;

public class TestReceiver extends BroadcastReceiver {
  @Override
  public void onReceive(Context context, Intent intent) {
    if (intent == null) {
      return;
    }

    if ("com.memfault.intent.action.LOG_METRIC".equals(intent.getAction())) {

      List<NumericAgg> numericAggs = asList(COUNT, NumericAgg.LATEST_VALUE, MAX, MEAN, MIN, SUM);
      List<StateAgg> stateAggs = asList(TIME_TOTALS, TIME_PER_HOUR, StateAgg.LATEST_VALUE);

      Reporting.report()
          .counter("reporting-maven-local-java-counter")
          .increment();
      Reporting.report()
          .sync(true)
          .record(true);
      Reporting.report()
          .successOrFailure("reporting-maven-local-java-sf")
          .record(false);
      Reporting.report()
          .distribution("reporting-maven-local-java-dist", numericAggs)
          .record(100.0);
      Reporting.report()
          .stringStateTracker("reporting-maven-local-java-sst", stateAggs)
          .state("stated");
      Reporting.report()
          .boolStateTracker("reporting-maven-local-java-bst", stateAggs)
          .state(false);
      Reporting.report()
          .stringProperty("reporting-maven-local-java-sp", true)
          .update("record");
      Reporting.report()
          .numberProperty("reporting-maven-local-java-np", true)
          .update(200L);
      Reporting.report()
          .event("reporting-maven-local-java-event", true, true)
          .add("evented");

      Reporting.startSession("reporting-maven-local-session");

      Distribution session = Reporting.session("reporting-maven-local-session")
          .distribution("reporting-maven-local-session-dist", numericAggs);
      session.record(2000L);
      session.record(8000L);

      Reporting.finishSession("reporting-maven-local-session");
    }
  }
}

File: MemfaultPackages/reporting-libs/clients/maven-kotlin/src/main/java/com/memfault/bort/reporting/TestReceiver.kt
package com.memfault.bort.reporting

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.memfault.bort.reporting.NumericAgg.COUNT
import com.memfault.bort.reporting.NumericAgg.LATEST_VALUE
import com.memfault.bort.reporting.NumericAgg.MAX
import com.memfault.bort.reporting.NumericAgg.MEAN
import com.memfault.bort.reporting.NumericAgg.MIN
import com.memfault.bort.reporting.NumericAgg.SUM
import com.memfault.bort.reporting.StateAgg.TIME_PER_HOUR
import com.memfault.bort.reporting.StateAgg.TIME_TOTALS

class TestReceiver : BroadcastReceiver() {
    override fun onReceive(
        context: Context,
        intent: Intent,
    ) {
        when (intent.action) {
            "com.memfault.intent.action.LOG_METRIC" -> {
                val numericAggs = listOf(COUNT, NumericAgg.LATEST_VALUE, MAX, MEAN, MIN, SUM)
                val stateAggs = listOf(TIME_TOTALS, TIME_PER_HOUR, StateAgg.LATEST_VALUE)

                Reporting.report()
                    .counter("reporting-maven-kotlin-counter")
                    .increment()
                Reporting.report()
                    .sync()
                    .record(true)
                Reporting.report()
                    .successOrFailure("reporting-maven-kotlin-sf")
                    .record(false)

                Reporting.report()
                    .distribution("reporting-maven-kotlin-dist", numericAggs)
                    .record(100)

                Reporting.report()
                    .distribution("reporting-maven-kotlin-dist1.min", listOf(LATEST_VALUE))
                    .record(1500)
                Reporting.report()
                    .distribution("reporting-maven-kotlin-dist1.mean", listOf(LATEST_VALUE))
                    .record(2024)
                Reporting.report()
                    .distribution("reporting-maven-kotlin-dist1.max", listOf(LATEST_VALUE))
                    .record(5000)

                Reporting.report()
                    .distribution("reporting-maven-kotlin-dist2.min", listOf(MIN))
                    .record(19)
                Reporting.report()
                    .distribution("reporting-maven-kotlin-dist2.mean", listOf(MEAN))
                    .record(24)
                Reporting.report()
                    .distribution("reporting-maven-kotlin-dist2.max", listOf(MAX))
                    .record(30)

                Reporting.report()
                    .stringStateTracker("reporting-maven-kotlin-sst", stateAggs)
                    .state("stated")
                Reporting.report()
                    .boolStateTracker("reporting-maven-kotlin-bst", stateAggs)
                    .state(false)
                Reporting.report()
                    .stringProperty("reporting-maven-kotlin-sp")
                    .update("record")
                Reporting.report()
                    .numberProperty("reporting-maven-kotlin-np")
                    .update(200L)
                Reporting.report()
                    .event("reporting-maven-kotlin-event", countInReport = true)
                    .add("evented")

                Reporting.startSession("reporting-maven-kotlin-session")
                val sessionDistribution = Reporting.session("reporting-maven-kotlin-session")
                    .distribution("reporting-maven-kotlin-session-dist", aggregations = listOf(MIN))
                sessionDistribution.record(10)
                sessionDistribution.record(20)
                Reporting.finishSession("reporting-maven-kotlin-session")
            }
        }
    }
}
